Failing to connect to elasticsearch

I am trying to use transporter to connect to an AWS Elasticsearch instance. I'm getting a panic error of the form: 

panic: Panic at 40: unable to determine version from .... bad status code: 403

I've tested the AWS credentials and URI by signing an Elasticsearch request with them and they seem to be working ok.

My Elasticsearch sink config is of the form:

var sink = elasticsearch({
    'uri': 'URI',
    'aws_access_key': 'AWS_ACCESS_KEY',
    'aws_access_secret': 'AWS_ACCESS_SECRET'
})

I'm using the latest (v0.4.0) darwin binary release.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -8,6 +8,8 @@ JS function to do it, fixed via [#401](https://github.com/compose/transporter/pu
 
 ### Bugfixes
 - when using an adaptor that was configured for flushing bulk messages based on an interval, it was possible for the bulk flush to error but not propagated back up to the pipeline, fixed via [#399](https://github.com/compose/transporter/pull/399)
+- if using AWS Elasticsearch service, the process to determine the version of elasticsearch failed,
+fixed via [#403](https://github.com/compose/transporter/pull/403)
 
 ## v0.4.0 [2017-08-15]
 

--- a/adaptor/elasticsearch/elasticsearch.go
+++ b/adaptor/elasticsearch/elasticsearch.go
@@ -89,17 +89,6 @@ func setupWriter(conf *Elasticsearch) (client.Writer, error) {
 		uri.Path = fmt.Sprintf("/%s", DefaultIndex)
 	}
 
-	hostsAndPorts := strings.Split(uri.Host, ",")
-	stringVersion, err := determineVersion(fmt.Sprintf("%s://%s", uri.Scheme, hostsAndPorts[0]), uri.User)
-	if err != nil {
-		return nil, err
-	}
-
-	v, err := version.NewVersion(stringVersion)
-	if err != nil {
-		return nil, client.VersionError{URI: conf.URI, V: stringVersion, Err: err.Error()}
-	}
-
 	timeout, err := time.ParseDuration(conf.Timeout)
 	if err != nil {
 		log.Debugf("failed to parse duration, %s, falling back to default timeout of 30s", conf.Timeout)
@@ -109,6 +98,21 @@ func setupWriter(conf *Elasticsearch) (client.Writer, error) {
 	httpClient := &http.Client{
 		Timeout:   timeout,
 		Transport: newTransport(conf.AWSAccessKeyID, conf.AWSAccessSecret),
+	}
+
+	hostsAndPorts := strings.Split(uri.Host, ",")
+	stringVersion, err := determineVersion(
+		fmt.Sprintf("%s://%s", uri.Scheme, hostsAndPorts[0]),
+		uri.User,
+		httpClient,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	v, err := version.NewVersion(stringVersion)
+	if err != nil {
+		return nil, client.VersionError{URI: conf.URI, V: stringVersion, Err: err.Error()}
 	}
 
 	for _, vc := range clients.Clients {
@@ -131,7 +135,7 @@ func setupWriter(conf *Elasticsearch) (client.Writer, error) {
 	return nil, client.VersionError{URI: conf.URI, V: stringVersion, Err: "unsupported client"}
 }
 
-func determineVersion(uri string, user *url.Userinfo) (string, error) {
+func determineVersion(uri string, user *url.Userinfo, httpClient *http.Client) (string, error) {
 	req, err := http.NewRequest(http.MethodGet, uri, nil)
 	if err != nil {
 		return "", err
@@ -141,7 +145,7 @@ func determineVersion(uri string, user *url.Userinfo) (string, error) {
 			req.SetBasicAuth(user.Username(), pwd)
 		}
 	}
-	resp, err := http.DefaultClient.Do(req)
+	resp, err := httpClient.Do(req)
 	if err != nil {
 		return "", client.ConnectError{Reason: uri}
 	}

